buildSrc/src/main/kotlin/pklKotlinLibrary.gradle.kts

```diff
@@ -1,9 +1,29 @@
+// https://youtrack.jetbrains.com/issue/KTIJ-19369
+@file:Suppress("DSL_SCOPE_VIOLATION")
+
+import io.gitlab.arturbosch.detekt.Detekt
+import io.gitlab.arturbosch.detekt.report.ReportMergeTask
+import org.gradle.accessors.dm.LibrariesForLibs
+
 plugins {
   id("pklJavaLibrary")
-
+  id("io.gitlab.arturbosch.detekt")
+  id("org.jetbrains.kotlinx.kover")
   kotlin("jvm")
 }
 
+// Build properties that control reporting.
+private val xmlReportingBuildProperty = "xmlReporting"
+private val sarifReportingBuildProperty = "sarifReporting"
+private val htmlReportingBuildProperty = "htmlReporting"
+
+// Resolved build properties for reporting within this project.
+val xmlReportingEnabled = findProperty(xmlReportingBuildProperty) == "true"
+val sarifReportingEnabled = findProperty(sarifReportingBuildProperty) == "true"
+val htmlReportingEnabled = findProperty(htmlReportingBuildProperty) == "true"
```

Comment:
XML, HTML, and SARIF reporting are all supported with Detekt, with merging enabled; each can be enabled with the properties above, all of which default to `false` in the root `gradle.properties`.